zulipterminal/core.py

```diff
@@ -492,6 +492,31 @@ def show_media_confirmation_popup(
             self, question, callback, location="center"
         )
 
+    def update_search_text_for_narrow(self, search_query: str) -> str:
```

Comment:
This looks more complete than the previous version :+1:

Note that all the internal code refers to the model data. This suggests the function belongs in the model.

Also note that we adapt the passed-in string as a return value. It's good to keep each function for one purpose, so leaving the string handling (eg. appending) for the query as something outside of this function.